src/com/xilinx/rapidwright/util/RunVivado.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2021-2022, Xilinx, Inc.
+ * Copyright (c) 2022-2023, Advanced Micro Devices, Inc.
+ * All rights reserved.
+ *
+ * Author: Zak Nafziger, Xilinx Research Labs.
+ *
+ * This file is part of RapidWright.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.xilinx.rapidwright.util;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Utility methods to run a tcl script through vivado and parse the stdout
+ * @author zakn
+ *
+ */
+public class RunVivado {
+    public static List<String> searchVivadoLog(List<String> log, String key) {
+        List<String> results = new ArrayList<>();
+        for(String l : log) {
+            if(l.contains(key)) {
+                results.add(l);
+            }
+        }
+        return results;
+    }
+
+    public static List<String> runVivadoTask(String runDir, String tclScript, boolean verbose) {
+        final String vivadoCmd = "vivado -mode batch -source " + tclScript;
+        System.out.println(vivadoCmd);
+
+        // set up the vivado process
+        Job j = new LocalJob();
+        j.setCommand(vivadoCmd);
+        j.setRunDir(runDir);
+        j.launchJob();
+
+        // run the vivado job
+        while (!j.isFinished()) {
+            if(verbose) {
+                System.out.println("Vivado running");
+            }
+            try {
+                Thread.sleep(5000);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+        }
```

Comment:
I'm not sure a `Job` here is the best tool here, since it does not output interactively to the screen -- I would prefer to see the progress on the screen, rather than wait until it terminates and then the entire output to appear. Consider using:
https://github.com/Xilinx/RapidWright/blob/45adb4f852b944dd3aaff14803edf8be8567cecd/src/com/xilinx/rapidwright/util/FileTools.java#L1553-L1560

FYI: There is a `JobQueue` class that takes care of this wait-until-done for you. Example:
https://github.com/Xilinx/RapidWright/blob/45adb4f852b944dd3aaff14803edf8be8567cecd/src/com/xilinx/rapidwright/interchange/GenerateInterchangeDevices.java#L45-L53
Typically, it's used for when there is more than one job being run in parallel.

Reply:
`FileTools.runCommand` intercepts std.out and std.err in order to print the output, and as a result I don't think there's an easy way to post process it (e.g. to parse the output of `report_route_status`). Perhaps I missed something obvious, but that's why I went with a `job`. I agree that printing the output live would be better though. I'll see if I can make that happen.

Reply:
Consider that Vivado has `-log` argument that allows you to override the output log file name (which gets written on top of printing to stdout). Even more specific is that `report_route_status` has a `-file` option which writes its output to a file (but looks like it doesn't print to screen?)

Reply:
Reading the vivado logfile could work. Right now I'm also using the `setRunDir()` method of `LocalJob` to ensure that output logs can be put in a `tempDir`. I'm not sure if there's a similar way to do this with `runCommand`, it seems like we'd just pollute the root directory with vivado logs otherwise. Although perhaps this isn't that big of an issue.